app/models/piece.rb

```diff
@@ -106,18 +106,35 @@ def is_obstructed?(destination_x, destination_y)
 		# moving diagonally left + down
 		(self.y_coordinate - 1).downto(destination_y).each do |y|
 			between_squares = game.pieces.where(x_coordinate: self.x_coordinate - 1, y_coordinate: y).first
-			if between_squares.present? 
+			if between_squares.present?
 				found = true
-				break 
+				break
 			end
 		end
 		found
 	end
 
+	# helper methods
 	def piece_at(x, y)
 		game.pieces.where(x_coordinate: x, y_coordinate: y).take
 	end
 
+  def update_coordinates(new_x, new_y)
+  	update(x_coordinate: new_x, y_coordinate: new_y)
+  end
+
+	def horizontal_move(x, y)
```

Comment:
do those methods always return a boolean? 
what gets returned in the case the condition is not satisfied?
I'd propose to add a '?' to the method name and a return false at the end to have all cases covered